doc/man3/SSL_set_async_callback.pod

```diff
@@ -0,0 +1,101 @@
+ int SSL_CTX_set_async_callback_arg(SSL_CTX *ctx, void *arg);
+ int SSL_set_async_callback(SSL *s, SSL_async_callback_fn callback);
+ int SSL_set_async_callback_arg(SSL *s, void *arg);
+ int SSL_get_async_status(SSL *s, int *status);
+ int SSL_clear_async_status(SSL *s);
+
+=head1 DESCRIPTION
+
+SSL_CTX_set_async_callback() sets asynchronous callback function, and all SSL
+sessions generated based on this SSL_CTX can get this callback, which will be
+automatically called when engine completed the cryptography operation to notify
+the application to resume the paused work flow.
+
+SSL_CTX_set_async_callback_arg() sets the callback argument.
+
+SSL_set_async_callback() allows a SSL application to set a callback in an
+asynchronous SSL session, so that when the engine completed a cryptography
+operation, the callback can be called to notify the application to resume the
+paused work flow.
+
+SSL_set_async_callback_arg() helps to set an argument for the SSL session when
+the above callback is called.
+
+SSL_get_async_status() returns the engine status. This function facilitates the
+communication from the engine to the application. During an SSL session,
+cryptographic operation would be dispatched to the engine. The engine status is
+very useful for an application to know if the engine has successfully dispatched
+it to hardware. If the engine does not support this additional callback method,
+"ASYNC_STATUS_UNSUPPORTED" will be returned.
+
+SSL_clear_async_status() helps SSL application to clear the engine status if the
+action has been taken already to avoid duplication handing.
+
+An example of the above functions would be the following.
+1. Application sets the async callback and callback data on an SSL connection
+by calling SSL_set_async_callback().
+2. Application makes an asynchronous SSL call
+3. OpenSSL submits the asynchronous request within the OpenSSL engine. If a
+retry occurs at this point then the status within the ASYNC_WAIT_CTX would be
+set and the async callback function would be called (goto Step 7).
+4. The OpenSSL engine pauses current job and return, so the application can
+continue processing other connections.
+5. At a future point in time (probably via a polling mechanism or via an
+interrupt) the engine will become aware that the asynchronous request has
+finished processing.
+6. The engine will call the application's callback passing the callback data as
+a parameter.
+7. The callback function should then run. Note: it is a requirement that the
+callback function is small and non-blocking as it will be run in the context of
+a polling mechanism or an interrupt.
+8. It is the application's responsibility via the callback function to schedule
+recalling the OpenSSL asynchronous function and to continue processing
+within the fibre.
+9. The callback function has the option to check the status returned via
+SSL_get_async_status() within the ASYNC_WAIT_CTX to determine whether a retry
```

Comment:
Remove the reference to ASYNC_WAIT_CTX here. We're not exposing that via the SSL object.